src/controllers/api-key.js

```diff
@@ -10,52 +10,184 @@
+  SERVER_ERROR,
+  STATUS_BAD_REQUEST,
+  STATUS_CREATED,
+  STATUS_FORBIDDEN,
+  STATUS_NO_CONTENT,
+  STATUS_UNAUTHORIZED,
+} from '../utils/constants.js';
+import { ApiKeyDto } from '../dto/api-key.js';
 
 /**
  * ApiKey Controller. Provides methods for managing API keys such as create, delete, and get.
+ * @param {Object} context - The context the universal serverless function.
+ * @param context.env - Environment details.
+ * @param context.env.API_KEY_CONFIGURATION - Configuration for the API key.
  * @returns {Object} ApiKey Controller
  * @constructor
  */
 function ApiKeyController(context) {
-  // eslint-disable-next-line no-unused-vars
-  const { log } = context;
+  const {
+    dataAccess, log, env, attributes, imsClient,
+  } = context;
+
+  let apiKeyConfiguration = {};
+  try {
+    apiKeyConfiguration = JSON.parse(env.API_KEY_CONFIGURATION);
+  } catch (error) {
+    log.error(`Failed to parse API Key configuration: ${error.message}`);
+  }
+
+  const { maxDomainsPerApiKey = 1, maxApiKeys = 3 } = apiKeyConfiguration;
   const HEADER_ERROR = 'x-error';
 
   function createErrorResponse(error) {
-    return createResponse({}, error.status, {
+    return createResponse({}, error.status || SERVER_ERROR, {
       [HEADER_ERROR]: error.message,
     });
   }
 
-  // eslint-disable-next-line no-unused-vars
+  /**
+   * Validate the request data.
+   * @param data
+   */
   function validateRequestData(data) {
-    // Not implemented yet
+    if (!isObject(data)) {
+      throw new ErrorWithStatusCode('Invalid request: missing application/json request data', STATUS_BAD_REQUEST);
+    }
+
+    if (!Array.isArray(data.features) || data.features.length === 0) {
+      throw new ErrorWithStatusCode('Invalid request: missing features in the request data', STATUS_BAD_REQUEST);
+    }
+
+    if (!Array.isArray(data.domains) || data.domains.length === 0) {
+      throw new ErrorWithStatusCode('Invalid request: missing domains in the request data', STATUS_BAD_REQUEST);
+    }
+
+    data.domains.forEach((url) => {
+      if (!isValidUrl(url)) {
+        throw new ErrorWithStatusCode(`Invalid request: ${url} is not a valid domain`, STATUS_BAD_REQUEST);
+      }
+    });
+
+    if (!hasText(data.name)) {
+      throw new ErrorWithStatusCode('Invalid request: missing name in the request data', STATUS_BAD_REQUEST);
+    }
+  }
+
+  /**
+   * Validate the IMS Org ID.
+   * @param imsOrgId
+   * @param imsUserToken
+   */
+  async function validateImsOrgId(imsOrgId, imsUserToken) {
+    if (!imsOrgId) {
+      throw new ErrorWithStatusCode('Missing x-ims-gw-org-id header', STATUS_UNAUTHORIZED);
+    }
+    const imsUserProfile = await imsClient.getImsUserProfile(imsUserToken);
+    log.debug(`Retrieved the IMS User Profile: ${JSON.stringify(imsUserProfile)}`);
+    const { organizations } = imsUserProfile;
+    if (!organizations.includes(imsOrgId)) {
+      throw new ErrorWithStatusCode('Invalid request: Organization not found', STATUS_UNAUTHORIZED);
```

Comment:
I was keeping the error message a bit vague to prevent bad actors. But, we could modify it to :
`Unable to find a reference to the Organization provided.`

Reply:
Works for me 👍